src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES(?, ?)";
+        List<Driver> drivers = car.getDrivers();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement creationStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            creationStatement.setString(1, car.getModel());
+            creationStatement.setLong(2, car.getManufacturer().getId());
+            creationStatement.executeUpdate();
+            ResultSet resultSet = creationStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            car.setDrivers(getDriversForCar(car.getId()));
+            insertDriversToCarDriversDB(car, drivers);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+    }
```

Comment:
```suggestion
        } catch (SQLException e) {
            throw new DataProcessingException("Couldn't create "
                    + car + ". ", e);
        }
        car.setDrivers(getDriversForCar(car.getId()));
        insertDriversToCarDriversDB(car, drivers);
        return car;
    }
```
you are opening an another connection inside those methods

<img width="1301" alt="image" src="https://user-images.githubusercontent.com/81781479/224697403-9905b385-223e-4345-9c39-60b466eabffb.png">